packages/components/src/Badge/Badge.component.js

```diff
@@ -8,60 +8,58 @@ import { DEFAULT_I18N } from '../translate';
 import Action from '../Actions/Action';
 import theme from './Badge.scss';
 
-function badgeClasses(tcStyle, onDelete) {
-	return classNames({
-		[theme['tc-badge']]: true,
-		'tc-badge': true,
-		[theme['tc-badge-outline']]: tcStyle === 'outline',
-		'tc-badge-outline': tcStyle === 'outline',
-		[theme['tc-badge-solid']]: tcStyle === 'solid',
-		'tc-badge-solid': tcStyle === 'solid',
-		[theme['tc-badge-with-icon']]: onDelete,
-		'tc-badge-with-icon': onDelete,
-	});
-}
-
-function badgeLabelClasses(onDelete) {
-	return classNames({
-		[theme['tc-badge-label']]: true,
-		'tc-badge-label': true,
-		[theme['tc-badge-label-with-icon']]: onDelete,
-		'tc-badge-label-with-icon': onDelete,
-	});
-}
-
-function renderDeleteIcon(onDelete, t) {
+function renderDeleteIcon(onClick, disabled, t) {
 	return (
 		<Action
 			label={t('BADGE_DELETE', { defaultValue: 'delete' })}
 			hideLabel
-			onClick={onDelete}
+			onClick={onClick}
+			disabled={disabled}
 			icon="talend-cross"
-			className={`${theme['tc-badge-delete-icon']} tc-badge-delete-icon`}
+			className={classNames('tc-badge-delete-icon', theme['tc-badge-delete-icon'])}
 		/>
 	);
 }
 
-function Badge({ label, tcStyle, onDelete, t }) {
+function Badge({ label, category, onDelete, onSelect, selected, disabled, t }) {
+	const containerClasses = classNames(
+		'tc-badge', theme['tc-badge'],
+		selected && ['tc-badge-selected', theme['tc-badge-selected']],
+		disabled && ['tc-badge-disabled', theme['tc-badge-disabled']],
+		!onDelete && ['tc-badge-readonly', theme['tc-badge-readonly']],
+	);
+	const badgeClasses = classNames('tc-badge-button', theme['tc-badge-button']);
+	const labelClasses = classNames('tc-badge-label', theme['tc-badge-label']);
+	const categoryClasses = classNames('tc-badge-category', theme['tc-badge-category']);
+
 	return (
-		<div className={badgeClasses(tcStyle, onDelete)}>
-			<span className={badgeLabelClasses(onDelete)}>
-				{label}
-			</span>
-			{onDelete && renderDeleteIcon(onDelete, t)}
+		<div className={containerClasses}>
+			<button className={badgeClasses} onClick={!disabled && onSelect} disabled={disabled}>
```

Comment:
Please avoid the `!disabled && onSelect` condition. It will create an empty onClick handler.

Reply:
onSelect is optional, so onClick can be empty even without this check